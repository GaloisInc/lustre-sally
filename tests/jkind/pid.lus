-- This is an example based on PID controller in AADL translated using
-- AGREE. There is a signle property which is provable at k = 23.
-- Increasing the value of max_accel makes k smaller and the proof
-- easier, and vice-versa for decreasing max_accel.

node H(inp : bool) returns (outp : bool);
let
  outp = (inp -> (inp and pre outp));
tel;

node abs(inp : real) returns (outp : real);
let
  outp = (if (inp < 0.0) then -inp else inp);
tel;

node AXL__abs(AXL__inp : real) returns (AXL__outp : real);
let
  AXL__outp = (if (AXL__inp < 0.0) then -AXL__inp else AXL__inp);
tel;

node _MAIN(
  AXL__Speed__val : real;
  Target_Speed__val : real;
  AXL__Actual_Tire_Direction__val : real;
  CNTRL__Actuator_Input__val : real;
  THROT__Actual__val : real;
  Target_Tire_Pitch__val : real
) returns (
);
var
  THROT__Actuator_Input__val : real;
  CNTRL__e_dot : real;
  CNTRL__e_int : real;
  CNTRL__e : real;
  CNTRL__u : real;
  max_accel : real;
  AXL__Target_Tire_Direction__val : real;
  Actual_Speed__val : real;
  CNTRL__Actual__val : real;
  CNTRL__Target__val : real;
  _TOTAL_COMP_HIST : bool;
  _SYSTEM_ASSUMP_HIST : bool;
  _CONTR_HIST_AXL : bool;
  _CONTR_HIST_CNTRL : bool;
  _CONTR_HIST_THROT : bool;
  _SYS_GUARANTEE_1 : bool;
let
  --%MAIN
  max_accel = 10.0;
  CNTRL__e = (CNTRL__Target__val - CNTRL__Actual__val);
  CNTRL__e_int = ((0.0 -> pre CNTRL__e) + CNTRL__e);
  CNTRL__e_dot = ((0.0 -> pre CNTRL__e) - CNTRL__e);
  CNTRL__u = (((0.2 * CNTRL__e) + (0.1 * CNTRL__e_dot)) + (0.1 * CNTRL__e_int));
  THROT__Actuator_Input__val = CNTRL__Actuator_Input__val;
  Actual_Speed__val = THROT__Actual__val;
  CNTRL__Actual__val = THROT__Actual__val;
  CNTRL__Target__val = Target_Speed__val;
  AXL__Target_Tire_Direction__val = Target_Tire_Pitch__val;
  _TOTAL_COMP_HIST = ((((true and (((AXL__abs(AXL__Target_Tire_Direction__val) <= 0.45) and (if ((AXL__abs(AXL__Target_Tire_Direction__val) > 0.2) and (AXL__Speed__val > 45.0)) then (AXL__Actual_Tire_Direction__val = 0.2) else (AXL__Actual_Tire_Direction__val = AXL__Target_Tire_Direction__val))) and (if ((AXL__abs(AXL__Target_Tire_Direction__val) > 0.2) and (AXL__Speed__val > 45.0)) then (AXL__Actual_Tire_Direction__val = 0.2) else (AXL__Actual_Tire_Direction__val = AXL__Target_Tire_Direction__val)))) and ((true and (CNTRL__Actuator_Input__val = CNTRL__u)) and (CNTRL__Actuator_Input__val = CNTRL__u))) and ((true and (THROT__Actual__val = ((0.0 -> pre THROT__Actual__val) + (0.1 * THROT__Actuator_Input__val)))) and (THROT__Actual__val = ((0.0 -> pre THROT__Actual__val) + (0.1 * THROT__Actuator_Input__val))))) -> ((((true and (((AXL__abs(AXL__Target_Tire_Direction__val) <= 0.45) and (if ((AXL__abs(AXL__Target_Tire_Direction__val) > 0.2) and (AXL__Speed__val > 45.0)) then (AXL__Actual_Tire_Direction__val = 0.2) else (AXL__Actual_Tire_Direction__val = AXL__Target_Tire_Direction__val))) and (if ((AXL__abs(AXL__Target_Tire_Direction__val) > 0.2) and (AXL__Speed__val > 45.0)) then (AXL__Actual_Tire_Direction__val = 0.2) else (AXL__Actual_Tire_Direction__val = AXL__Target_Tire_Direction__val)))) and ((true and (CNTRL__Actuator_Input__val = CNTRL__u)) and (CNTRL__Actuator_Input__val = CNTRL__u))) and ((true and (THROT__Actual__val = ((0.0 -> pre THROT__Actual__val) + (0.1 * THROT__Actuator_Input__val)))) and (THROT__Actual__val = ((0.0 -> pre THROT__Actual__val) + (0.1 * THROT__Actuator_Input__val))))) and pre _TOTAL_COMP_HIST));
  _SYSTEM_ASSUMP_HIST = ((((Target_Speed__val >= 0.0) and (Target_Speed__val < 150.0)) and true) -> ((((Target_Speed__val >= 0.0) and (Target_Speed__val < 150.0)) and true) and pre _SYSTEM_ASSUMP_HIST));
  _CONTR_HIST_AXL = ((((AXL__abs(AXL__Target_Tire_Direction__val) <= 0.45) and (if ((AXL__abs(AXL__Target_Tire_Direction__val) > 0.2) and (AXL__Speed__val > 45.0)) then (AXL__Actual_Tire_Direction__val = 0.2) else (AXL__Actual_Tire_Direction__val = AXL__Target_Tire_Direction__val))) and (if ((AXL__abs(AXL__Target_Tire_Direction__val) > 0.2) and (AXL__Speed__val > 45.0)) then (AXL__Actual_Tire_Direction__val = 0.2) else (AXL__Actual_Tire_Direction__val = AXL__Target_Tire_Direction__val))) -> ((((AXL__abs(AXL__Target_Tire_Direction__val) <= 0.45) and (if ((AXL__abs(AXL__Target_Tire_Direction__val) > 0.2) and (AXL__Speed__val > 45.0)) then (AXL__Actual_Tire_Direction__val = 0.2) else (AXL__Actual_Tire_Direction__val = AXL__Target_Tire_Direction__val))) and (if ((AXL__abs(AXL__Target_Tire_Direction__val) > 0.2) and (AXL__Speed__val > 45.0)) then (AXL__Actual_Tire_Direction__val = 0.2) else (AXL__Actual_Tire_Direction__val = AXL__Target_Tire_Direction__val))) and pre _CONTR_HIST_AXL));
  _CONTR_HIST_CNTRL = (((true and (CNTRL__Actuator_Input__val = CNTRL__u)) and (CNTRL__Actuator_Input__val = CNTRL__u)) -> (((true and (CNTRL__Actuator_Input__val = CNTRL__u)) and (CNTRL__Actuator_Input__val = CNTRL__u)) and pre _CONTR_HIST_CNTRL));
  _CONTR_HIST_THROT = (((true and (THROT__Actual__val = ((0.0 -> pre THROT__Actual__val) + (0.1 * THROT__Actuator_Input__val)))) and (THROT__Actual__val = ((0.0 -> pre THROT__Actual__val) + (0.1 * THROT__Actuator_Input__val)))) -> (((true and (THROT__Actual__val = ((0.0 -> pre THROT__Actual__val) + (0.1 * THROT__Actuator_Input__val)))) and (THROT__Actual__val = ((0.0 -> pre THROT__Actual__val) + (0.1 * THROT__Actuator_Input__val)))) and pre _CONTR_HIST_THROT));
  _SYS_GUARANTEE_1 = ((_SYSTEM_ASSUMP_HIST and _TOTAL_COMP_HIST) => (abs((Actual_Speed__val - (0.0 -> pre Actual_Speed__val))) < max_accel));
  --%PROPERTY _SYS_GUARANTEE_1;
tel;
