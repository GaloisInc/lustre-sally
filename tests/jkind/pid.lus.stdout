[Lustre] Loading settings from: "pid.lus.settings"
Sally Model
===========

(define-state-type
   S
   (
     (abs Real)
     (inp Real)
     (|AXL__Actual_Tire_Direction__val| Real)
     (|AXL__Speed__val| Real)
     (|AXL__Target_Tire_Direction__val| Real)
     (|AXL__abs:10| Real)
     (|AXL__abs:11| Real)
     (|AXL__abs:1| Real)
     (|AXL__abs:2| Real)
     (|AXL__abs:3| Real)
     (|AXL__abs:4| Real)
     (|AXL__abs:5| Real)
     (|AXL__abs:6| Real)
     (|AXL__abs:7| Real)
     (|AXL__abs:8| Real)
     (|AXL__abs:9| Real)
     (|AXL__abs| Real)
     (|AXL__inp:10| Real)
     (|AXL__inp:11| Real)
     (|AXL__inp:1| Real)
     (|AXL__inp:2| Real)
     (|AXL__inp:3| Real)
     (|AXL__inp:4| Real)
     (|AXL__inp:5| Real)
     (|AXL__inp:6| Real)
     (|AXL__inp:7| Real)
     (|AXL__inp:8| Real)
     (|AXL__inp:9| Real)
     (|AXL__inp| Real)
     (|Actual_Speed__val| Real)
     (|CNTRL__Actual__val| Real)
     (|CNTRL__Actuator_Input__val| Real)
     (|CNTRL__Target__val| Real)
     (|CNTRL__e_dot| Real)
     (|CNTRL__e_int| Real)
     (|CNTRL__e| Real)
     (|CNTRL__u| Real)
     (|THROT__Actual__val| Real)
     (|THROT__Actuator_Input__val| Real)
     (|Target_Speed__val| Real)
     (|Target_Tire_Pitch__val| Real)
     (|_CONTR_HIST_AXL:init| Bool)
     (|_CONTR_HIST_AXL| Bool)
     (|_CONTR_HIST_CNTRL:init| Bool)
     (|_CONTR_HIST_CNTRL| Bool)
     (|_CONTR_HIST_THROT:init| Bool)
     (|_CONTR_HIST_THROT| Bool)
     (|_SYSTEM_ASSUMP_HIST:init| Bool)
     (|_SYSTEM_ASSUMP_HIST| Bool)
     (|_SYS_GUARANTEE_1| Bool)
     (|_TOTAL_COMP_HIST:init| Bool)
     (|_TOTAL_COMP_HIST| Bool)
     (|gal-initializing| Bool)
     (|init_pre_Actual_Speed__val:init| Bool)
     (|init_pre_Actual_Speed__val| Real)
     (|init_pre_CNTRL__e:1:init| Bool)
     (|init_pre_CNTRL__e:1| Real)
     (|init_pre_CNTRL__e:init| Bool)
     (|init_pre_CNTRL__e| Real)
     (|init_pre_THROT__Actual__val:1:init| Bool)
     (|init_pre_THROT__Actual__val:1| Real)
     (|init_pre_THROT__Actual__val:2:init| Bool)
     (|init_pre_THROT__Actual__val:2| Real)
     (|init_pre_THROT__Actual__val:3:init| Bool)
     (|init_pre_THROT__Actual__val:3| Real)
     (|init_pre_THROT__Actual__val:4:init| Bool)
     (|init_pre_THROT__Actual__val:4| Real)
     (|init_pre_THROT__Actual__val:5:init| Bool)
     (|init_pre_THROT__Actual__val:5| Real)
     (|init_pre_THROT__Actual__val:6:init| Bool)
     (|init_pre_THROT__Actual__val:6| Real)
     (|init_pre_THROT__Actual__val:7:init| Bool)
     (|init_pre_THROT__Actual__val:7| Real)
     (|init_pre_THROT__Actual__val:init| Bool)
     (|init_pre_THROT__Actual__val| Real)
     (|max_accel| Real)
     (|pre_Actual_Speed__val| Real)
     (|pre_CNTRL__e:1| Real)
     (|pre_CNTRL__e| Real)
     (|pre_THROT__Actual__val:1| Real)
     (|pre_THROT__Actual__val:2| Real)
     (|pre_THROT__Actual__val:3| Real)
     (|pre_THROT__Actual__val:4| Real)
     (|pre_THROT__Actual__val:5| Real)
     (|pre_THROT__Actual__val:6| Real)
     (|pre_THROT__Actual__val:7| Real)
     (|pre_THROT__Actual__val| Real)
     (|pre__CONTR_HIST_AXL| Bool)
     (|pre__CONTR_HIST_CNTRL| Bool)
     (|pre__CONTR_HIST_THROT| Bool)
     (|pre__SYSTEM_ASSUMP_HIST| Bool)
     (|pre__TOTAL_COMP_HIST| Bool))
   (
     (|AXL__Actual_Tire_Direction__val| Real)
     (|AXL__Speed__val| Real)
     (|CNTRL__Actuator_Input__val| Real)
     (|THROT__Actual__val| Real)
     (|Target_Speed__val| Real)
     (|Target_Tire_Pitch__val| Real)))
(define-transition-system
   TS
   S
   (and
      (= |gal-initializing| true)
      (= |init_pre_CNTRL__e:1:init| true)
      (= |init_pre_CNTRL__e:init| true)
      (= |init_pre_THROT__Actual__val:7:init| true)
      (= |init_pre_THROT__Actual__val:6:init| true)
      (= |init_pre_THROT__Actual__val:5:init| true)
      (= |init_pre_THROT__Actual__val:4:init| true)
      (= |_TOTAL_COMP_HIST:init| true)
      (= |_SYSTEM_ASSUMP_HIST:init| true)
      (= |_CONTR_HIST_AXL:init| true)
      (= |_CONTR_HIST_CNTRL:init| true)
      (= |init_pre_THROT__Actual__val:3:init| true)
      (= |init_pre_THROT__Actual__val:2:init| true)
      (= |init_pre_THROT__Actual__val:1:init| true)
      (= |init_pre_THROT__Actual__val:init| true)
      (= |_CONTR_HIST_THROT:init| true)
      (= |init_pre_Actual_Speed__val:init| true))
   (and
      (= next.|gal-initializing| false)
      (= next.|AXL__Speed__val| input.|AXL__Speed__val|)
      (= next.|Target_Speed__val| input.|Target_Speed__val|)
      (= next.|AXL__Actual_Tire_Direction__val| input.|AXL__Actual_Tire_Direction__val|)
      (= next.|CNTRL__Actuator_Input__val| input.|CNTRL__Actuator_Input__val|)
      (= next.|THROT__Actual__val| input.|THROT__Actual__val|)
      (= next.|Target_Tire_Pitch__val| input.|Target_Tire_Pitch__val|)
      (= next.|THROT__Actuator_Input__val| next.|CNTRL__Actuator_Input__val|)
      (= next.|CNTRL__Actual__val| next.|THROT__Actual__val|)
      (= next.|CNTRL__Target__val| next.|Target_Speed__val|)
      (=
         next.|CNTRL__e|
         (- next.|CNTRL__Target__val| next.|CNTRL__Actual__val|))
      (= next.|pre_CNTRL__e:1| state.|CNTRL__e|)
      (ite
         state.|init_pre_CNTRL__e:1:init|
         (= next.|init_pre_CNTRL__e:1| 0.0)
         (= next.|init_pre_CNTRL__e:1| next.|pre_CNTRL__e:1|))
      (= next.|init_pre_CNTRL__e:1:init| false)
      (=
         next.|CNTRL__e_dot|
         (- next.|init_pre_CNTRL__e:1| next.|CNTRL__e|))
      (= next.|pre_CNTRL__e| state.|CNTRL__e|)
      (ite
         state.|init_pre_CNTRL__e:init|
         (= next.|init_pre_CNTRL__e| 0.0)
         (= next.|init_pre_CNTRL__e| next.|pre_CNTRL__e|))
      (= next.|init_pre_CNTRL__e:init| false)
      (=
         next.|CNTRL__e_int|
         (+ next.|init_pre_CNTRL__e| next.|CNTRL__e|))
      (=
         next.|CNTRL__u|
         (+
            (+
               (*
                  (/ 1 5)
                  next.|CNTRL__e|)
               (*
                  (/ 1 10)
                  next.|CNTRL__e_dot|))
            (*
               (/ 1 10)
               next.|CNTRL__e_int|)))
      (= next.|max_accel| 10.0)
      (= next.|AXL__Target_Tire_Direction__val| next.|Target_Tire_Pitch__val|)
      (= next.|Actual_Speed__val| next.|THROT__Actual__val|)
      (= next.|AXL__inp:11| next.|AXL__Target_Tire_Direction__val|)
      (=
         next.|AXL__abs:11|
         (ite
            (< next.|AXL__inp:11| 0.0)
            (- next.|AXL__inp:11|)
            next.|AXL__inp:11|))
      (= next.|AXL__inp:10| next.|AXL__Target_Tire_Direction__val|)
      (=
         next.|AXL__abs:10|
         (ite
            (< next.|AXL__inp:10| 0.0)
            (- next.|AXL__inp:10|)
            next.|AXL__inp:10|))
      (= next.|AXL__inp:9| next.|AXL__Target_Tire_Direction__val|)
      (=
         next.|AXL__abs:9|
         (ite
            (< next.|AXL__inp:9| 0.0)
            (- next.|AXL__inp:9|)
            next.|AXL__inp:9|))
      (= next.|AXL__inp:8| next.|AXL__Target_Tire_Direction__val|)
      (=
         next.|AXL__abs:8|
         (ite
            (< next.|AXL__inp:8| 0.0)
            (- next.|AXL__inp:8|)
            next.|AXL__inp:8|))
      (= next.|AXL__inp:7| next.|AXL__Target_Tire_Direction__val|)
      (=
         next.|AXL__abs:7|
         (ite
            (< next.|AXL__inp:7| 0.0)
            (- next.|AXL__inp:7|)
            next.|AXL__inp:7|))
      (= next.|AXL__inp:6| next.|AXL__Target_Tire_Direction__val|)
      (=
         next.|AXL__abs:6|
         (ite
            (< next.|AXL__inp:6| 0.0)
            (- next.|AXL__inp:6|)
            next.|AXL__inp:6|))
      (= next.|pre_THROT__Actual__val:7| state.|THROT__Actual__val|)
      (ite
         state.|init_pre_THROT__Actual__val:7:init|
         (= next.|init_pre_THROT__Actual__val:7| 0.0)
         (= next.|init_pre_THROT__Actual__val:7| next.|pre_THROT__Actual__val:7|))
      (= next.|init_pre_THROT__Actual__val:7:init| false)
      (= next.|pre_THROT__Actual__val:6| state.|THROT__Actual__val|)
      (ite
         state.|init_pre_THROT__Actual__val:6:init|
         (= next.|init_pre_THROT__Actual__val:6| 0.0)
         (= next.|init_pre_THROT__Actual__val:6| next.|pre_THROT__Actual__val:6|))
      (= next.|init_pre_THROT__Actual__val:6:init| false)
      (= next.|pre_THROT__Actual__val:5| state.|THROT__Actual__val|)
      (ite
         state.|init_pre_THROT__Actual__val:5:init|
         (= next.|init_pre_THROT__Actual__val:5| 0.0)
         (= next.|init_pre_THROT__Actual__val:5| next.|pre_THROT__Actual__val:5|))
      (= next.|init_pre_THROT__Actual__val:5:init| false)
      (= next.|pre_THROT__Actual__val:4| state.|THROT__Actual__val|)
      (ite
         state.|init_pre_THROT__Actual__val:4:init|
         (= next.|init_pre_THROT__Actual__val:4| 0.0)
         (= next.|init_pre_THROT__Actual__val:4| next.|pre_THROT__Actual__val:4|))
      (= next.|init_pre_THROT__Actual__val:4:init| false)
      (= next.|pre__TOTAL_COMP_HIST| state.|_TOTAL_COMP_HIST|)
      (ite
         state.|_TOTAL_COMP_HIST:init|
         (=
            next.|_TOTAL_COMP_HIST|
            (and
               true
               (<=
                  next.|AXL__abs:11|
                  (/ 9 20))
               (ite
                  (and
                     (<
                        (/ 1 5)
                        next.|AXL__abs:10|)
                     (< 45.0 next.|AXL__Speed__val|))
                  (=
                     next.|AXL__Actual_Tire_Direction__val|
                     (/ 1 5))
                  (= next.|AXL__Actual_Tire_Direction__val| next.|AXL__Target_Tire_Direction__val|))
               (ite
                  (and
                     (<
                        (/ 1 5)
                        next.|AXL__abs:9|)
                     (< 45.0 next.|AXL__Speed__val|))
                  (=
                     next.|AXL__Actual_Tire_Direction__val|
                     (/ 1 5))
                  (= next.|AXL__Actual_Tire_Direction__val| next.|AXL__Target_Tire_Direction__val|))
               true
               (= next.|CNTRL__Actuator_Input__val| next.|CNTRL__u|)
               (= next.|CNTRL__Actuator_Input__val| next.|CNTRL__u|)
               true
               (=
                  next.|THROT__Actual__val|
                  (+
                     next.|init_pre_THROT__Actual__val:7|
                     (*
                        (/ 1 10)
                        next.|THROT__Actuator_Input__val|)))
               (=
                  next.|THROT__Actual__val|
                  (+
                     next.|init_pre_THROT__Actual__val:6|
                     (*
                        (/ 1 10)
                        next.|THROT__Actuator_Input__val|)))))
         (=
            next.|_TOTAL_COMP_HIST|
            (and
               true
               (<=
                  next.|AXL__abs:8|
                  (/ 9 20))
               (ite
                  (and
                     (<
                        (/ 1 5)
                        next.|AXL__abs:7|)
                     (< 45.0 next.|AXL__Speed__val|))
                  (=
                     next.|AXL__Actual_Tire_Direction__val|
                     (/ 1 5))
                  (= next.|AXL__Actual_Tire_Direction__val| next.|AXL__Target_Tire_Direction__val|))
               (ite
                  (and
                     (<
                        (/ 1 5)
                        next.|AXL__abs:6|)
                     (< 45.0 next.|AXL__Speed__val|))
                  (=
                     next.|AXL__Actual_Tire_Direction__val|
                     (/ 1 5))
                  (= next.|AXL__Actual_Tire_Direction__val| next.|AXL__Target_Tire_Direction__val|))
               true
               (= next.|CNTRL__Actuator_Input__val| next.|CNTRL__u|)
               (= next.|CNTRL__Actuator_Input__val| next.|CNTRL__u|)
               true
               (=
                  next.|THROT__Actual__val|
                  (+
                     next.|init_pre_THROT__Actual__val:5|
                     (*
                        (/ 1 10)
                        next.|THROT__Actuator_Input__val|)))
               (=
                  next.|THROT__Actual__val|
                  (+
                     next.|init_pre_THROT__Actual__val:4|
                     (*
                        (/ 1 10)
                        next.|THROT__Actuator_Input__val|)))
               next.|pre__TOTAL_COMP_HIST|)))
      (= next.|_TOTAL_COMP_HIST:init| false)
      (= next.|pre__SYSTEM_ASSUMP_HIST| state.|_SYSTEM_ASSUMP_HIST|)
      (ite
         state.|_SYSTEM_ASSUMP_HIST:init|
         (=
            next.|_SYSTEM_ASSUMP_HIST|
            (and
               (<= 0.0 next.|Target_Speed__val|)
               (< next.|Target_Speed__val| 150.0)
               true))
         (=
            next.|_SYSTEM_ASSUMP_HIST|
            (and
               (<= 0.0 next.|Target_Speed__val|)
               (< next.|Target_Speed__val| 150.0)
               true
               next.|pre__SYSTEM_ASSUMP_HIST|)))
      (= next.|_SYSTEM_ASSUMP_HIST:init| false)
      (= next.|AXL__inp:5| next.|AXL__Target_Tire_Direction__val|)
      (=
         next.|AXL__abs:5|
         (ite
            (< next.|AXL__inp:5| 0.0)
            (- next.|AXL__inp:5|)
            next.|AXL__inp:5|))
      (= next.|AXL__inp:4| next.|AXL__Target_Tire_Direction__val|)
      (=
         next.|AXL__abs:4|
         (ite
            (< next.|AXL__inp:4| 0.0)
            (- next.|AXL__inp:4|)
            next.|AXL__inp:4|))
      (= next.|AXL__inp:3| next.|AXL__Target_Tire_Direction__val|)
      (=
         next.|AXL__abs:3|
         (ite
            (< next.|AXL__inp:3| 0.0)
            (- next.|AXL__inp:3|)
            next.|AXL__inp:3|))
      (= next.|AXL__inp:2| next.|AXL__Target_Tire_Direction__val|)
      (=
         next.|AXL__abs:2|
         (ite
            (< next.|AXL__inp:2| 0.0)
            (- next.|AXL__inp:2|)
            next.|AXL__inp:2|))
      (= next.|AXL__inp:1| next.|AXL__Target_Tire_Direction__val|)
      (=
         next.|AXL__abs:1|
         (ite
            (< next.|AXL__inp:1| 0.0)
            (- next.|AXL__inp:1|)
            next.|AXL__inp:1|))
      (= next.|AXL__inp| next.|AXL__Target_Tire_Direction__val|)
      (=
         next.|AXL__abs|
         (ite
            (< next.|AXL__inp| 0.0)
            (- next.|AXL__inp|)
            next.|AXL__inp|))
      (= next.|pre__CONTR_HIST_AXL| state.|_CONTR_HIST_AXL|)
      (ite
         state.|_CONTR_HIST_AXL:init|
         (=
            next.|_CONTR_HIST_AXL|
            (and
               (<=
                  next.|AXL__abs:5|
                  (/ 9 20))
               (ite
                  (and
                     (<
                        (/ 1 5)
                        next.|AXL__abs:4|)
                     (< 45.0 next.|AXL__Speed__val|))
                  (=
                     next.|AXL__Actual_Tire_Direction__val|
                     (/ 1 5))
                  (= next.|AXL__Actual_Tire_Direction__val| next.|AXL__Target_Tire_Direction__val|))
               (ite
                  (and
                     (<
                        (/ 1 5)
                        next.|AXL__abs:3|)
                     (< 45.0 next.|AXL__Speed__val|))
                  (=
                     next.|AXL__Actual_Tire_Direction__val|
                     (/ 1 5))
                  (= next.|AXL__Actual_Tire_Direction__val| next.|AXL__Target_Tire_Direction__val|))))
         (=
            next.|_CONTR_HIST_AXL|
            (and
               (<=
                  next.|AXL__abs:2|
                  (/ 9 20))
               (ite
                  (and
                     (<
                        (/ 1 5)
                        next.|AXL__abs:1|)
                     (< 45.0 next.|AXL__Speed__val|))
                  (=
                     next.|AXL__Actual_Tire_Direction__val|
                     (/ 1 5))
                  (= next.|AXL__Actual_Tire_Direction__val| next.|AXL__Target_Tire_Direction__val|))
               (ite
                  (and
                     (<
                        (/ 1 5)
                        next.|AXL__abs|)
                     (< 45.0 next.|AXL__Speed__val|))
                  (=
                     next.|AXL__Actual_Tire_Direction__val|
                     (/ 1 5))
                  (= next.|AXL__Actual_Tire_Direction__val| next.|AXL__Target_Tire_Direction__val|))
               next.|pre__CONTR_HIST_AXL|)))
      (= next.|_CONTR_HIST_AXL:init| false)
      (= next.|pre__CONTR_HIST_CNTRL| state.|_CONTR_HIST_CNTRL|)
      (ite
         state.|_CONTR_HIST_CNTRL:init|
         (=
            next.|_CONTR_HIST_CNTRL|
            (and
               true
               (= next.|CNTRL__Actuator_Input__val| next.|CNTRL__u|)
               (= next.|CNTRL__Actuator_Input__val| next.|CNTRL__u|)))
         (=
            next.|_CONTR_HIST_CNTRL|
            (and
               true
               (= next.|CNTRL__Actuator_Input__val| next.|CNTRL__u|)
               (= next.|CNTRL__Actuator_Input__val| next.|CNTRL__u|)
               next.|pre__CONTR_HIST_CNTRL|)))
      (= next.|_CONTR_HIST_CNTRL:init| false)
      (= next.|pre_THROT__Actual__val:3| state.|THROT__Actual__val|)
      (ite
         state.|init_pre_THROT__Actual__val:3:init|
         (= next.|init_pre_THROT__Actual__val:3| 0.0)
         (= next.|init_pre_THROT__Actual__val:3| next.|pre_THROT__Actual__val:3|))
      (= next.|init_pre_THROT__Actual__val:3:init| false)
      (= next.|pre_THROT__Actual__val:2| state.|THROT__Actual__val|)
      (ite
         state.|init_pre_THROT__Actual__val:2:init|
         (= next.|init_pre_THROT__Actual__val:2| 0.0)
         (= next.|init_pre_THROT__Actual__val:2| next.|pre_THROT__Actual__val:2|))
      (= next.|init_pre_THROT__Actual__val:2:init| false)
      (= next.|pre_THROT__Actual__val:1| state.|THROT__Actual__val|)
      (ite
         state.|init_pre_THROT__Actual__val:1:init|
         (= next.|init_pre_THROT__Actual__val:1| 0.0)
         (= next.|init_pre_THROT__Actual__val:1| next.|pre_THROT__Actual__val:1|))
      (= next.|init_pre_THROT__Actual__val:1:init| false)
      (= next.|pre_THROT__Actual__val| state.|THROT__Actual__val|)
      (ite
         state.|init_pre_THROT__Actual__val:init|
         (= next.|init_pre_THROT__Actual__val| 0.0)
         (= next.|init_pre_THROT__Actual__val| next.|pre_THROT__Actual__val|))
      (= next.|init_pre_THROT__Actual__val:init| false)
      (= next.|pre__CONTR_HIST_THROT| state.|_CONTR_HIST_THROT|)
      (ite
         state.|_CONTR_HIST_THROT:init|
         (=
            next.|_CONTR_HIST_THROT|
            (and
               true
               (=
                  next.|THROT__Actual__val|
                  (+
                     next.|init_pre_THROT__Actual__val:3|
                     (*
                        (/ 1 10)
                        next.|THROT__Actuator_Input__val|)))
               (=
                  next.|THROT__Actual__val|
                  (+
                     next.|init_pre_THROT__Actual__val:2|
                     (*
                        (/ 1 10)
                        next.|THROT__Actuator_Input__val|)))))
         (=
            next.|_CONTR_HIST_THROT|
            (and
               true
               (=
                  next.|THROT__Actual__val|
                  (+
                     next.|init_pre_THROT__Actual__val:1|
                     (*
                        (/ 1 10)
                        next.|THROT__Actuator_Input__val|)))
               (=
                  next.|THROT__Actual__val|
                  (+
                     next.|init_pre_THROT__Actual__val|
                     (*
                        (/ 1 10)
                        next.|THROT__Actuator_Input__val|)))
               next.|pre__CONTR_HIST_THROT|)))
      (= next.|_CONTR_HIST_THROT:init| false)
      (= next.|pre_Actual_Speed__val| state.|Actual_Speed__val|)
      (ite
         state.|init_pre_Actual_Speed__val:init|
         (= next.|init_pre_Actual_Speed__val| 0.0)
         (= next.|init_pre_Actual_Speed__val| next.|pre_Actual_Speed__val|))
      (= next.|init_pre_Actual_Speed__val:init| false)
      (=
         next.inp
         (- next.|Actual_Speed__val| next.|init_pre_Actual_Speed__val|))
      (=
         next.abs
         (ite
            (< next.inp 0.0)
            (- next.inp)
            next.inp))
      (=
         next.|_SYS_GUARANTEE_1|
         (=>
            (and next.|_SYSTEM_ASSUMP_HIST| next.|_TOTAL_COMP_HIST|)
            (< next.abs next.|max_accel|)))))

(query
   TS
   (or |gal-initializing| |_SYS_GUARANTEE_1|))

[Lustre] Validating properties:
[Lustre] Property _SYS_GUARANTEE_1...
  [Sally][Valid] using 23 past states
[Lustre] Summary:
  [Valid] 1
  [Unknown] 0
  [Invalid] 0
[Lustre] Model status: [Valid]
